draft-ietf-taps-transport-security.md

```diff
@@ -249,11 +251,63 @@ interface or address changes.
 
 - Server: the peer responsible for responding to a session initiation.
 
+# Security Features
+
+In this section, we enumerate Security Features exposed by protocols discussed in the
+remainder of this document. Protocol security properties that are unrelated to the API surface
+exposed by such protocols, such as client or server identity hiding, are not listed 
+here as features.
+
+- Forward-secure key establishment: Cryptographic key establishment with forward secure properties.
+
+- Cryptographic algorithm negotiation: Negotiate support of protocol algorithms, including: encryption, 
+hash, MAC (PRF), and digital signature algorithms.
+
+- Stateful and stateless cross-connection session resumption: Connection establishment without needing 
+to complete an entirely new handshake. 
+
+- Peer authentication (certificate, raw public key, pre-shared key, or EAP-based): Peer authentication using 
+select or protocol-specific mechanisms.
+
+- Mutual authentication: Connection establishment wherein both endpoints are authenticated.
+
+- Record (channel or datagram) confidentiality and integrity: Encryption and authentication of 
+application plaintext bytes sent between peers over a channel or in individual datagrams.
+
+- Partial record confidentiality: Encryption of some portion of records.
+
+- Optional record integrity: Optional authentication of certain records.
+
+- Record replay prevention: Protocol detection and defense against record replays, e.g., due 
+to in-network retransmissions.
+
+- Application-layer feature negotiation: Securely negotiate application-specific functionality.
+
+- Early data support (starting with TLS 1.3): Transmission of application data prior to connection 
+(handshake) establishment.
+
+- Connection mobility: Connection continuation in the presence of 5-tuple changes beneath the 
+secure transport protocol, e.g., due to NAT rebindings.
+
+- Application-layer authentication delegation: Out-of-band peer authentication performed by 
+applications outside of the connection establishment.
+
+- Out-of-order record receipt: Processing of records received out-of-order.
+
+- DoS mitigation (cookie or puzzle based): Peer DoS mitigation via explicit proof of origin (cookie) or
+work mechanisms (puzzles).
+
+- Connection re-keying: In-band cryptographic handshake re-keying.
+
+- Optional length-hiding and anti-amplification padding: Protocol-drive record padding aimed at hiding
```

Comment:
Is "optional" here (and elsewhere) redundant? It isn't the case that all of these features are provided by all protocols.

Reply:
I don't think so. A protocol may support the feature, but not always use it.

Reply:
Eh, I agree that Optional doesn't need to be in the title for this. You can just mention it's optional in the description.